lib/repl.js

```diff
@@ -249,7 +250,17 @@ function REPLServer(prompt,
         }
         script = vm.createScript(code, {
           filename: file,
-          displayErrors: true
+          displayErrors: true,
+          async resolveDynamicImport(specifier, scriptOrModule) {
+            const loader = await require('internal/process/esm_loader')
+              .loaderPromise;
+
+            const filename = scriptOrModule.filename;
+            const referrer = scriptOrModule.url ||
+              getURLFromFilePath(filename === 'repl' ?
+                path.join(process.cwd(), filename) : filename).href;
+            return loader.import(specifier, referrer);
```

Comment:
How come this can work without any `isNamespace` checks then?

Reply:
it doesn't work, it's wrong and needs a test :D